daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -675,12 +667,11 @@ private[lf] final class Compiler(
       tmplId: TypeConName,
       tmpl: Template,
       fields: ImmArray[(FieldName, Type)],
-      predPids: List[PackageId],
       choice: TemplateChoice,
   ): (t.SDefinitionRef, SDefinition) =
     topLevelFunction3(t.SoftTemplateChoiceDefRef(tmplId, choice.name)) {
       (cidPos, choiceArgPos, tokenPos, env) =>
-        val casting = Casting.Soft(tmplId, fields, predPids)
```

Comment:
Are we absolutely sure we want to completely strip the notion of an _upgrades_ relation between `PackageId`?

Reply:
As an alternative, we could abstract this notion as a new class/trait `UpgradabilityRelation`.;
pass a value of this type through the code, to which we could defer question such as `isUpgrade` or `isDowngrade`.

And then have it return always _true_ (for now) if we want.